### PR TITLE
Finished the base navbar layout and Register SignIn function

### PR DESCRIPTION
**Pull request by ExploreNcrack**

### CODE REVIEW

File: GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+{
+    public class AccountController : Controller
+    {
+        private readonly UserManager<ApplicationUser> userManager;
+        private readonly SignInManager<ApplicationUser> signInManager;
+
+        public AccountController(UserManager<ApplicationUser> userManager,
+            SignInManager<ApplicationUser> signInManager)
+        {
+            this.userManager = userManager;
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs]: C# convention, don't use `this.xxx` to assign injected services.
Directly assign it as `_userManager = userManager`, using "_xxx" is also a convention
try to avoid using `this.` to assign any properties/variable within the same class

File: GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            this.signInManager = signInManager;
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs]: same as above

File: GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        }
+
+        // GET: /<controller>/
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs]: change to ---> `// GET: /Account/Register`

File: GunplaOnlineShop/GunplaOnlineShop/Data/Migrations/20200604025414_tryagain.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace GunplaOnlineShop.Data.Migrations
+{
+    public partial class tryagain : Migration
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Data/Migrations/20200604025414_tryagain.cs]: empty migration file?  Up and Down methods are empty, delete this migration file

File: GunplaOnlineShop/GunplaOnlineShop/Views/Home/Index.cshtml [hunk 5]
```diff
@@ -1,8 +1,8 @@
 ﻿@{
-    ViewData["Title"] = "Home Page";
+    ViewData["Title"] = "Hello Home";
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Views/Home/Index.cshtml]: This is the Title of the page which will display as the name of the tab, 
image
please use a proper name "xxx gunpla shop"

File: GunplaOnlineShop/GunplaOnlineShop/Views/Shared/_Layout.cshtml [hunk 6]
```diff
@@ -1,34 +1,88 @@
...
+                            <a class="nav-item" asp-controller="account" asp-action="Profile">
+                                @User.Identity.Name
+                            </a>
+                        </li>
+                    }
+                    else
+                    {
                         <li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Index">Home</a>
+                            <a class="nav-link" asp-controller="account" asp-action="register">SignUp</a>
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Views/Shared/_Layout.cshtml]: it is better to keep it as `“Account”` and `"Register"`

File: GunplaOnlineShop/GunplaOnlineShop/Views/Shared/_Layout.cshtml [hunk 7]
```diff
@@ -1,34 +1,88 @@
...
                         </li>
                         <li class="nav-item">
-                            <a class="nav-link text-dark" asp-area="" asp-controller="Home" asp-action="Privacy">Privacy</a>
+                            <a class="nav-link" asp-controller="account" asp-action="login">SignIn</a>
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Views/Shared/_Layout.cshtml]: same as above

File: GunplaOnlineShop/GunplaOnlineShop/appsettings.Development.example.json [hunk 8]
```diff
@@ -1,25 +0,0 @@
-{
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/appsettings.Development.example.json]: don't delete this configuration file, this is for other people who can use to copy paste and configure their own `appsettings.Development.json` file and run it on their own machine

**Review by ExploreNcrack — COMMENTED** [review]
some convention reviewed and more review will be coming

File: GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+
+        [HttpPost]
+        public async Task<IActionResult> logout()
+        {
+            await signInManager.SignOutAsync();
+            return RedirectToAction("index", "home");
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> Register(RegisterViewModel model)
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs:44]: `post` `Register` action method should be right below the `get`, it is better to keep them close together in the code

File: GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+        public IActionResult Register()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> logout()
+        {
+            await signInManager.SignOutAsync();
+            return RedirectToAction("index", "home");
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs]: `Index` and `Home`

File: GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        [HttpPost]
+        public async Task<IActionResult> Login(LoginViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var result = await signInManager.PasswordSignInAsync(model.Email, model.Password, model.RememberMe, false);
+
+                if (result.Succeeded)
+                {
+                    return RedirectToAction("index", "home");
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs]: `Index` and `Home`

File: GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+                }
+                else
+                {
+                    ModelState.AddModelError(string.Empty,"Failed to Login");
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Controllers/AccountController.cs]: "Invalid Login Attempt" sounds better?

File: GunplaOnlineShop/GunplaOnlineShop/Views/Account/Register.cshtml [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+}
+<link href="~/css/signin-form.css" rel="stylesheet" />
+<div class="signup-form">
+    <form method="post" class="form-horizontal">
+        <div class="col-xs-8 col-xs-offset-4">
+            <h2>Sign Up</h2>
+        </div>
+        <div asp-validation-summary="All" class="text-danger"></div>
+        <div class="form-group">
+            <label asp-for="Email" class="control-label col-xs-4">Email Address</label>
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Views/Account/Register.cshtml]: No need to type `Email Address` here
Use the `[Display(Name = "Email Address")]` data annotation attribute to decorate the property combine with `asp-for`, it will auto generate the html element with corresponding attributes

File: GunplaOnlineShop/GunplaOnlineShop/Views/Account/Register.cshtml [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+            <div class="col-xs-8">
+                <input asp-for="Email" class="form-control">
+            </div>
+            <span asp-validation-for="Email" class="text-danger"></span>
+        </div>
+        <div class="form-group">
+            <label asp-for="Password" class="control-label col-xs-4">Password</label>
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Views/Account/Register.cshtml]: Same as above

File: GunplaOnlineShop/GunplaOnlineShop/Views/Account/Register.cshtml [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+            <div class="col-xs-8">
+                <input asp-for="Password" class="form-control">
+            </div>
+            <span asp-validation-for="Password" class="text-danger"></span>
+        </div>
+        <div class="form-group">
+            <label asp-for="ConfirmPassword" class="control-label col-xs-4">Confirm Password</label>
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Views/Account/Register.cshtml]: Same as above

File: GunplaOnlineShop/GunplaOnlineShop/ViewModels/RegisterViewModel.cs [hunk 16]
```diff
@@ -9,11 +9,13 @@ namespace GunplaOnlineShop.ViewModels
     public class RegisterViewModel
     {
         [Required]
+        [Display(Name = "Email Address")]
         [DataType(DataType.EmailAddress)]
         [EmailAddress]
         public string Email { get; set; }
 
         [Required]
+        [Display(Name = "Password")]
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/ViewModels/RegisterViewModel.cs]: if you want to display the text the same as the property name, no need to use `[Display(Name = "Password")]`, by default, it will display as the property name  in `<label asp-for="xxx">`

File: GunplaOnlineShop/GunplaOnlineShop/ViewModels/LoginViewModel.cs [hunk 17]
```diff
@@ -9,11 +9,13 @@ namespace GunplaOnlineShop.ViewModels
     public class LoginViewModel
     {
         [Required]
+        [Display(Name = "Email Address")]
         [DataType(DataType.EmailAddress)]
         [EmailAddress]
         public string Email { get; set; }
 
         [Required]
+        [Display(Name = "Password")]
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/ViewModels/LoginViewModel.cs]: same as below and any where else

File: GunplaOnlineShop/GunplaOnlineShop/Views/Shared/_Layout.cshtml [hunk 18]
```diff
@@ -6,12 +6,11 @@
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>@ViewData["Title"] - GunplaOnlineShop</title>
-    <link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.min.css" />
-    <script src="~/lib/jquery/dist/jquery.min.js"></script>
-    <script src ="~/lib/bootstrap/dist/js/bootstrap.min.js"></script>
+    <link href="~/lib/bootstrap/dist/css/bootstrap.css" rel="stylesheet" />
+    <script src="//ajax.googleapis.com/ajax/libs/jquery/1.11.0/jquery.min.js"></script>
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/Views/Shared/_Layout.cshtml]: src should be starting with `src="https://xxxxxx/x.js"`, same with any where else

File: GunplaOnlineShop/GunplaOnlineShop/GunplaOnlineShop.csproj [hunk 19]
```diff
@@ -4,6 +4,14 @@
     <TargetFramework>netcoreapp3.1</TargetFramework>
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (ExploreNcrack) [inline GunplaOnlineShop/GunplaOnlineShop/GunplaOnlineShop.csproj:7]: delete this? if it is not necessary
